cirq/google/calibration/phased_fsim.py

```diff
@@ -203,9 +217,25 @@ def override_by(self, other: 'PhasedFSimCharacterization') -> 'PhasedFSimCharact
 )
 
 
-class PhasedFSimCalibrationOptions(abc.ABC):
+class PhasedFSimCalibrationOptions(abc.ABC, Generic[RequestT]):
     """Base class for calibration-specific options passed together with the requests."""
 
+    @abc.abstractmethod
+    def create_phased_fsim_request(
+        self,
+        pairs: Tuple[Tuple[Qid, Qid], ...],
+        gate: Gate,
+    ) -> RequestT:
```

Comment:
Are you sure you need this generic? I'm not sure about thus but I think they changed something in recent Python/mypy because in the code I'm writing recently, I can freely specify just a base class here and mypy doesn't complain.

Reply:
I certainly tried that first. I think I ran into trouble keeping the backwards-compatible `prepare_floquet_...` functions have the same return type. In any event, the generic keeps more type information with the resulting object.